common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisFlightEventManager.java

```diff
@@ -0,0 +1,277 @@
+                this.operator.getLevel().playSound(null, TardisArchitectureHandler.DESKTOP_CENTER_POS, SoundEvents.MINECART_INSIDE, SoundSource.BLOCKS, 1000f, 0.1f);
+            }
+
+            scaleForDanger = 0.7f;
+        }
+
+        if (dangerZoneSecondsPast(30)) {
+            if (operator.getLevel().getGameTime() % (3 * 20) == 0) {
+                playCloisterBell();
+            }
+
+            scaleForDanger = 1f;
+        }
+
+        if (dangerZoneSecondsPast(60)) {
+            for (Player player : this.operator.getLevel().players()) {
+                MobEffectInstance mobEffectInstance = new MobEffectInstance(MobEffects.DARKNESS, 10, 0, false, false);
+                player.addEffect(mobEffectInstance);
+            }
+
+            scaleForDanger = 1.75f;
+        }
+
+        this.dangerZoneShakeScale = scaleForDanger;
+
+
+        // The requests are too great, the TARDIS needs to crash.
+        if (this.requiredDangerZoneRequests >= 10) {
+            this.operator.getControlManager().crash();
+            this.isWaitingForControlResponse = false;
+            this.isInDangerZone = false;
+            this.ticksInTheDangerZone = 0;
+            this.dangerZoneShakeScale = 3f;
+        }
+    }
+
+    public void respondToWaitingControl(ControlEntity entity, ConsoleControl control) {
+        // Assign a cooldown between the controls determined by stats.
+        this.controlRequestCooldown = getControlRequestCooldown();
+
+        // Increment the number of control responses
+        // If we're in the danger zone, we don't want to advance the normal flight. We'd rather force the player to get out of it first.
+        if (isInDangerZone()) {
+            this.dangerZoneResponses++;
+        } else {
+            this.controlResponses++;
+        }
+
+        this.isWaitingForControlResponse = false;
+
+        if (this.controlResponses == this.requiredControlRequests) {
+            operator.getLevel().playSound(null, entity.blockPosition(), SoundEvents.PLAYER_LEVELUP, SoundSource.AMBIENT, 10, 1);
+        } else {
+            if (this.isEventInComboTime()) {
+                float pitch = 1.25f * getPercentComplete();
+                operator.getLevel().playSound(null, entity.blockPosition(), SoundEvents.EXPERIENCE_ORB_PICKUP, SoundSource.AMBIENT, 10, pitch);
+            }
+        }
+
+    }
+
+    private boolean dangerZoneSecondsPast(int seconds) {
+        return (this.ticksInTheDangerZone >= seconds * 20);
+    }
+
+    public float getPercentComplete() {
+        return (float)this.controlResponses / (float)this.requiredControlRequests;
+    }
+
+    private void playCloisterBell() {
+        this.operator.getLevel().playSound(null, TardisArchitectureHandler.DESKTOP_CENTER_POS, SoundEvents.BELL_BLOCK, SoundSource.BLOCKS, 1000f, 0.1f);
+        this.operator.getLevel().playSound(null, TardisArchitectureHandler.DESKTOP_CENTER_POS, SoundEvents.BELL_BLOCK, SoundSource.BLOCKS, 1000f, 0.1f);
```

Comment:
For loop this. Any lines of code that are exactly the same should not follow each other